src/core/radio/ble_secure.cpp

```diff
@@ -74,7 +76,13 @@ Error BleSecure::Start(ConnectCallback aConnectHandler, ReceiveCallback aReceive
     mMtuSize = kInitialMtuSize;
 
     SuccessOrExit(error = otPlatBleEnable(&GetInstance()));
+
+    SuccessOrExit(error = otPlatBleGetAdvertisementBuffer(&GetInstance(), &advertisementData));
+    SuccessOrExit(error = mTcatAgent.GetAdvertisementData(advertisementLen, advertisementData));
+    VerifyOrExit(advertisementData != nullptr, error = kErrorFailed);
+    SuccessOrExit(error = otPlatBleGapAdvSetData(&GetInstance(), advertisementData, advertisementLen));
```

Comment:
Getting the buffer from the platform works well in this case.

Alternatively, we could require `otPlatBleGapAdvSetData` to copy the provided data for the platform layer to retain and use. This would allow us to simplify the implementation here by using a local array.

Both approaches seem viable to me.

Reply:
Ok then let's leave it as is.